python/dgl/distgnn/partition/libra_partition.py

```diff
@@ -0,0 +1,577 @@
+    save_tensors(node_feat_file, g.ndata)
+    save_graphs(part_graph_file, [g])
+    print("Graph saved successfully !!")
+    
+
+    
+def load_proteins(dataset):
+    part_dir = dataset
+    graph_file = os.path.join(part_dir + "/graph.dgl")
+
+    if not os.path.exists("proteins.mtx"):
+        download_proteins()
+    if not os.path.exists(graph_file):
+        g = proteins_mtx2dgl()
+        save(g, dataset)
+    
+    graph = load_graphs(graph_file)[0][0]
+    return graph
+
+
+class Args:
+    def __init__(self, dataset):
+        self.dataset = dataset
+            
+                
+def leastload(weights_array):
+    result = np.where(weights_array == np.amin(weights_array))
+    return random.choice(result[0])
+
+
+def vertex_cut_partition(num_community, dataset, prefix):
+    """
+    Performs vertex-cut based grpah partitioning
+    Parameters
+    ----------
+    num_community : Number of partitions to create
+    dataset : Input graph name to partition
+    prefix : Output location
+
+    Output
+    ------
+    Creates X partition folder as XCommunities (say, X=2, so, 2Communities)
+    XCommunities contains communityZ.txt file per parition Z
+    Each such file contains list of edges assigned to that partition.
+
+    """
+    args = Args(dataset)
+    print("Input dataset: ", args.dataset)
+    if args.dataset == 'ogbn-products':
+        print("Loading ogbn-products")
+        G,_ = load_ogb('ogbn-products')
+    elif args.dataset == 'ogbn-papers100M':
+        print("Loading ogbn-papers100M")
+        G,_ = load_ogb('ogbn-papers100M')
+    elif args.dataset == 'proteins':
+        G = load_proteins('proteins')
+    elif args.dataset == 'ogbn-arxiv':
+        print("Loading ogbn-arxiv")
+        G, _ = load_ogb('ogbn-arxiv')
+    else:
+        try:
+            G = load_data(args)[0]
+        except:
+            raise DGLError("Error: Dataset {} not found !!!".format(dataset))
+
+    print("Done loading the graph.", flush=True)
+    
+    N_n = G.number_of_nodes()  # number of nodes
+    N_c = num_community     ## number of partitions/communities
+    N_e = G.number_of_edges()
+    community_list = [[] for i in range(N_c)]
+    
+    in_d = G.in_degrees()
+    out_d = G.out_degrees()
+    node_degree = in_d + out_d
+    edgenum_unassigned = node_degree.clone()
+    replication_list = []
+        
+    u_t,v_t = G.edges()
+    weight_ = th.ones(u_t.shape[0], dtype=th.float32)
+    community_weights = th.zeros(N_c, dtype=th.float32)
+
+    self_loop = 0
+    for i in range(len(u_t)):
+        if u_t[i] == v_t[i]:
+            self_loop += 1
+
+    print("#self loops in the dataset: ", self_loop)
+
+    del G
+
+    ## call to C/C++ code
+    out = th.zeros(u_t.shape[0], dtype=th.int32)
+    libra_vertex_cut(N_c, node_degree, edgenum_unassigned, community_weights,
+                     u_t, v_t, weight_, out, N_n, N_e, prefix)
+        
+    return int(community_weights.max())
+
+
+
+if __name__ == "__main__":
```

Comment:
I agree. The standalone functionality is unnecessary. Right now both parts of this script use built-in functions to load the datasets. Given a generic load function for dataset loading, it will be easier to write the data loading part of the script. Let me know if have something in mind, any suggestions. Or we push that to the next immediate PR on partitioning.

Reply:
Speaking of the Python interface itself, I think the best option will be providing a `libra_partition` function that takes in a single `DGLGraph` as well as a path to save all the files to.  This way you can write your standalone script by simply invoking the `libra_partition` function, while allowing users to still use Libra with their own graphs.